openedx/features/content_type_gating/tests/test_access.py

```diff
@@ -505,3 +505,223 @@ def test_access_based_on_conditional_content(self):
+        course = CourseFactory.create(run='test', display_name='test')
+        CourseModeFactory.create(course_id=course.id, mode_slug='audit')
+        CourseModeFactory.create(course_id=course.id, mode_slug='verified')
+        blocks_dict = {}
+        with self.store.bulk_operations(course.id):
+            blocks_dict['chapter'] = ItemFactory.create(
+                parent=course,
+                category='chapter',
+                display_name='Week 1'
+            )
+            blocks_dict['sequential'] = ItemFactory.create(
+                parent=blocks_dict['chapter'],
+                category='sequential',
+                display_name='Lesson 1'
+            )
+            blocks_dict['vertical'] = ItemFactory.create(
+                parent=blocks_dict['sequential'],
+                category='vertical',
+                display_name='Lesson 1 Vertical - Unit 1'
+            )
+        return {
+            'course': course,
+            'blocks': blocks_dict,
+        }
+
+    def test_single_denied(self):
+        ''' Single graded problem should show error '''
+        course = self._create_course()
+        blocks_dict = course['blocks']
+        CourseEnrollmentFactory.create(
+            user=self.user,
+            course_id=course['course'].id,
+            mode='audit'
+        )
+        blocks_dict['graded_1'] = ItemFactory.create(
+            parent=blocks_dict['vertical'],
+            category='problem',
+            display_name='graded_problem',
+            graded=True,
+        )
+        _assert_block_is_gated(
+            block=blocks_dict['graded_1'],
+            user_id=self.user.id,
+            course=course['course'],
+            is_gated=True,
+            request_factory=self.request_factory,
+        )
+
+    def test_double_denied(self):
+        ''' First graded problem should show message, second shouldn't '''
+        course = self._create_course()
+        blocks_dict = course['blocks']
+        blocks_dict['graded_1'] = ItemFactory.create(
+            parent=blocks_dict['vertical'],
+            category='problem',
+            display_name='graded_problem',
+            graded=True,
+        )
+        blocks_dict['graded_2'] = ItemFactory.create(
+            parent=blocks_dict['vertical'],
+            category='problem',
+            display_name='graded_problem',
+            graded=True,
+        )
+        CourseEnrollmentFactory.create(
+            user=self.user,
+            course_id=course['course'].id,
+            mode='audit'
+        )
+        _assert_block_is_gated(
+            block=blocks_dict['graded_1'],
+            user_id=self.user.id,
+            course=course['course'],
+            is_gated=True,
+            request_factory=self.request_factory,
+        )
+        _assert_block_is_gated(
+            block=blocks_dict['graded_2'],
+            user_id=self.user.id,
+            course=course['course'],
+            is_gated=False,
```

Comment:
imo here and in all the other spots in this PR where the blocks after the first block don't have a content gate it would be good to check that the block has the correct value (what is the correct value? empty div? empty string?) rather than asserting that the block does not contain the string 'content-paywall'.  
for instance, if the full problem showed up here rather than being hidden, this test would still pass

Reply:
That's a solid point. Will refactor the checks.

Reply:
@MatthewPiatetsky Tests have been updated